### PR TITLE
Dont fail on extension change only

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes https://github.com/dotnet/docs-actions/issues/12

~~Hard to confirm the correctness of the change without running against an actual PR. Consider creating a new branch, merge the PR to it, and try the change in dotnet/docs before this gets merged into main. So if this change has any bugs/issues, it doesn't affect PRs coming to dotnet/docs.~~ No need to do that. It was easy to confirm on my fork of dotnet/docs.

### CODE REVIEW

**Comment by Youssef1313** [comment]
@IEvangelist @gewarren for review.

File: actions/docs-verifier/src/ActionRunner/Program.cs [hunk 1]
```diff
@@ -23,20 +23,46 @@
...
+        }
+        else if (file.EndsWith(".md", StringComparison.OrdinalIgnoreCase))
+        {
+            strippedFile = file.Substring(0, file.Length - ".md".Length);
+            return true;
+        }
+
+        strippedFile = file;
+        return false;
+    }
```
Comment (IEvangelist) [inline actions/docs-verifier/src/ActionRunner/Program.cs:65]:
```suggestion
    static bool TryStripYmlOrMarkdownExtension(string file, out string strippedFile)
    {
        var (subFile, result) = Path.GetExtension(file) switch
        {
            string ext when ext is ".yml" or ".yaml" or ".md" =>
                (file.Substring(0, file.Length - ext.Length), true),

            _ => (file, false)
        };
        
        strippedFile = subFile;
        return result;
    }
```

File: actions/docs-verifier/src/ActionRunner/Program.cs [hunk 2]
```diff
@@ -23,20 +23,46 @@
...
             returnCode++;
         }
     }
+
+    static bool IsExtensionChangeOnly(string file1, string file2)
+    {
+        return TryStripYmlOrMarkdownExtension(file1, out string strippedFile1) &&
+            TryStripYmlOrMarkdownExtension(file2, out string strippedFile2) &&
+            strippedFile1.Equals(strippedFile2, StringComparison.OrdinalIgnoreCase);
+    }
```
Comment (IEvangelist) [inline actions/docs-verifier/src/ActionRunner/Program.cs:48]:
```suggestion
    static bool IsExtensionChangeOnly(string file1, string file2) =>
        TryStripYmlOrMarkdownExtension(file1, out string strippedFile1) &&
        TryStripYmlOrMarkdownExtension(file2, out string strippedFile2) &&
        strippedFile1.Equals(strippedFile2, StringComparison.OrdinalIgnoreCase);
```

**Review by IEvangelist — APPROVED** [review]
A few nits

File: actions/docs-verifier/src/ActionRunner/Program.cs (hunk 2 above)
Comment (Youssef1313) [reply actions/docs-verifier/src/ActionRunner/Program.cs:48]: @IEvangelist `yaml` doesn't seem supported by docs build. See:

image

vs

image

Build shows a preview link for `yml`, but not `yaml`. I'll adjust this and apply the suggestion.

File: actions/docs-verifier/.editorconfig [hunk 3]
```diff
@@ -23,9 +23,6 @@ dotnet_diagnostic.CA2007.severity = none
 # IDE0044: Add readonly modifier
 dotnet_style_readonly_field = true
 
-# Default severity for analyzer diagnostics with category 'Style'
-dotnet_analyzer_diagnostic.category-Style.severity = warning
```
Comment (Youssef1313) [inline actions/docs-verifier/.editorconfig:27]: I think this is an unnecessary noise.

Comment (IEvangelist) [reply actions/docs-verifier/.editorconfig:27]: Then remove it, this is your PR after all 😄

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**